cpp/terrain/terrain_search.h

```diff
@@ -0,0 +1,71 @@
+// Copyright 2015-2015 the openage authors. See copying.md for legal info.
+
+#ifndef OPENAGE_TERRAIN_TERRAIN_SEARCH_H_
+#define OPENAGE_TERRAIN_TERRAIN_SEARCH_H_
+
+#include <functional>
+#include <memory>
+#include <queue>
+#include <unordered_set>
+#include <vector>
+
+#include "../coord/tile.h"
+
+namespace openage {
+
+class Terrain;
+class TerrainObject;
+
+std::shared_ptr<TerrainObject> find_near(const TerrainObject &start, 
+                                         std::function<bool(const TerrainObject &)> found);
+
+constexpr coord::tile_delta const neigh_tile[] = {
+	{0, 1},
+	{0,  -1},
+	{1,  0},
+	{-1,  0}
```

Comment:
if you're gonna use a completely irregular number of spaces, at least make it so that stuff is aligned.